src/shared/awsContext.ts

```diff
@@ -41,6 +44,7 @@ export class DefaultAwsContext implements AwsContext {
     public readonly onDidChangeContext: vscode.Event<ContextChangeEventsArgs>
     private readonly _onDidChangeContext: vscode.EventEmitter<ContextChangeEventsArgs>
     private shim?: CredentialsShim
+    public lastTouchedRegion: string | undefined
```

Comment:
Yeah, you're right. I just checked and I never set it back to undefined. I actually check to make sure the new region isn't undefined before setting it so this could never happen.